File: loteria.js
module.exports = (req, res) => {
    const { minimo = 1, maximo = 50, quantidade = 5, comb_size = 3 } = req.query;

    // Gerar número aleatório entre min e max
    const gerarNumeroAleatorio = (min, max) => {
        min = Math.ceil(min);
        max = Math.floor(max);
        return Math.floor(Math.random() * (max - min + 1)) + min;
    };

    // Gerar números aleatórios
    const gerarNumerosAleatorios = (min, max, quant) => {
        let resultados = [];
        for (let i = 0; i < quant; i++) {
            resultados.push(gerarNumeroAleatorio(min, max));
        }
        return resultados;
    };

    // Gerar combinações
    const combinar = (arr, combSize) => {
        const combinations = [];
        const combine = (temp, left, right) => {
            if (combSize === temp.length) {
                combinations.push(temp.slice());
                return;
            }
            for (let i = left; i <= right; i++) {
                temp.push(arr[i]);
                combine(temp, i + 1, right);
                temp.pop();
            }
        };
        combine([], 0, arr.length - 1);
        return combinations;
    };

    // Processar dados
    const numeros = gerarNumerosAleatorios(parseInt(minimo), parseInt(maximo), parseInt(quantidade));
    const combinacoes = combinar(numeros, parseInt(comb_size));

    // Enviar resposta
    res.status(200).json({ numeros, combinacoes });
};

File: loteria.py
import random
import itertools

class Loteria:
    def gerar_numero_aleat(self, minimo, maximo, quantidade):
        return [random.randint(minimo, maximo) for _ in range(quantidade)]

    def gerar_combinacoes(self, numeros, tamanho_comb):
        return list(itertools.combinations(numeros, tamanho_comb))

# Exemplo de uso
loteria = Loteria()

# Gerar 5 números aleatórios entre 1 e 50
numeros_aleatorios = loteria.gerar_numero_aleat(1, 50, 5)
print("Números Aleatórios:", numeros_aleatorios)

# Gerar todas as combinações possíveis de 3 números a partir dos números gerados
combinacoes = loteria.gerar_combinacoes(numeros_aleatorios, 3)
print("Combinacoes possíveis:", list(combinacoes))
